### ISSUE TITLE
ContextMenu

### ISSUE THREAD

**Issue by juavol** [FlaUI/FlaUI#340]
I am struggling with the ContextMenu of a WPF application. The the ContextMenu itself is on a treeview somewhere down the visual tree of the main window. I can open the context menu by simulating a right-click on the corresponding control somewhat like this:
`var result = Content.FindAllDescendants(ConditionFactory.ByClassName("TreeViewItem")).FirstOrDefault(...);
result.RightClick();
Wait.UntilInputIsProcessed(Timeout);
Wait.UntilResponsive(result, Timeout);
`
After that, I would like to further use the context menu, specifically to invoke one of this items.
When I try
`var m = mainWindow.ContextMenu.FindFirstChild(...)`
then this will work most of the time, but sometimes fails with ContextMenu being null. However, even in cases where it fails, the menu does appear on screen.
I even tried to insert a Thread.Sleep() after the RightClick() but it does not seem to solve the problem either.

Is this a known issue or is there a better way to open the context menu and invoke one of its items.

**Comment by Roemer** [comment]
Strange that it sometimes works. Seems like a timing issue but you tried already with a sleep. Could you also try with a `Retry.WhileNull?`

**Comment by Roemer** [comment]
Did you try with a `Retry` or found another solution?

**Comment by tomhornyak** [comment]
I to am having this problem - and I am using a Retry.WhileNull. I set a 10 second timeout and I can see the context menu, but the retry fails. In my case also it is intermediate.

**Comment by kerberk** [comment]
> I to am having this problem - and I am using a Retry.WhileNull. I set a 10 second timeout and I can see the context menu, but the retry fails. In my case also it is intermediate.

Did you find a solution for this in the meantime?

**Comment by tomhornyak** [comment]
No. I gave up on ui testing. Using unit tests now. Sent from my iPhoneOn May 20, 2024, at 10:30 AM, Karsten Kerber ***@***.***> wrote:﻿

I to am having this problem - and I am using a Retry.WhileNull. I set a 10 second timeout and I can see the context menu, but the retry fails. In my case also it is intermediate.

Did you find a solution for this in the meantime?

**Comment by MakszCsepes** [comment]
Hi. I am facing the same issue as @tomhornyak . Tried Retry.WhileNull but no result. The context menu appears to be always null on the window object...

Will appreciate any suggestions.

**Comment by jeffhuckins** [comment]
I'm having the same issue. Why is nobody from the FlaUI team addressing this? I have a .NET 6 Desktop application.

**Comment by jdh28** [comment]
It seems that in more recent versions of .NET, the context menu's name is no longer "DropDown" - it's just an empty string.
